src/main/java/seedu/address/ui/UpcomingTuitionListPanel.java

```diff
@@ -77,33 +75,55 @@ public int compare(Tuition tuition1, Tuition tuition2) {
     }
 
     /**
-     * Filters all tuition 1 week from today.
-     */
-    private void filterOneWeekTuitionSessions(ObservableList<Tuition> tuitionList,
-                                              ObservableList<Tuition> finalTuitionList) {
-        LocalDate today = LocalDateTime.now().toLocalDate();
-        LocalDate weekToday = today.plusWeeks(1);
-        for (Tuition tuition : tuitionList) {
-            LocalDate sessionDate = tuition.getSession().getSessionDate().getDate();
-            if (today.compareTo(sessionDate) <= 0 && sessionDate.compareTo(weekToday) < 0) {
-                finalTuitionList.add(tuition);
-            }
-        }
-    }
-
-    /**
-     * Populates the tuitionList with {@code Tuition} for all sessions in the studentList.
+     * Populates the tuitionList with {@code Tuition} for all sessions in the studentList between today and
+     * the day after tomorrow.
      */
     private void populateTuitionList(ObservableList<Student> studentList, ObservableList<Tuition> tuitionList) {
+
+        SessionDate today = new SessionDate(LocalDateTime.now());
+        SessionDate tomorrow = new SessionDate(LocalDateTime.now().plusDays(1));
+        SessionDate dayAfterTomorrow = new SessionDate(LocalDateTime.now().plusDays(2));
+
         for (int i = 0; i < studentList.size(); i++) {
             Student currStudent = studentList.get(i);
             for (int j = 0; j < studentList.get(i).getListOfSessions().size(); j++) {
                 Session currSession = currStudent.getListOfSessions().get(j);
-                tuitionList.add(new Tuition(currStudent, currSession, i, j));
+                if (currSession instanceof RecurringSession) {
+                    RecurringSession recurringSession = (RecurringSession) currSession;
+
+                    if (!recurringSession.endBefore(today)) {
+                        if (recurringSession.hasSessionOnDate(today)) {
+                            tuitionList.add(new Tuition(currStudent,
+                                    recurringSession.buildSessionOnDate(today.getDate()), i, j));
+                        }
+                        if (recurringSession.hasSessionOnDate(tomorrow)) {
+                            tuitionList.add(new Tuition(currStudent,
+                                    recurringSession.buildSessionOnDate(tomorrow.getDate()), i, j));
+                        }
+                        if (recurringSession.hasSessionOnDate(dayAfterTomorrow)) {
+                            tuitionList.add(new Tuition(currStudent,
+                                    recurringSession.buildSessionOnDate(dayAfterTomorrow.getDate()), i, j));
+                        }
+                    }
+                } else {
```

Comment:
Bring else clause to top.
if (recurringSession.endBefore(today)) {
...
break;
}

Reply:
And if you bother to, do a recurringSession.startAfter(today) {
break;
} also